packages/sitecore-jss/src/data-fetcher.ts

```diff
@@ -36,33 +37,17 @@ export class ResponseError extends Error {
   }
 }
 
-/**
- * @param {HttpResponse<T>} response the response to check
- * @throws {ResponseError} if response code is not ok
- */
-export function checkStatus<T>(response: HttpResponse<T>) {
-  if (response.status >= 200 && response.status < 300) {
-    return response;
-  }
-
-  const error = new ResponseError(response.statusText, response);
-  throw error;
-}
-
 /**
  * @param {string} url the URL to request; may include query string
  * @param {HttpDataFetcher} fetcher the fetcher to use to perform the request
  * @param {ParsedUrlQueryInput} params the query string parameters to send with the request
  */
-export function fetchData<T>(
+export async function fetchData<T>(
```

Comment:
`async` isn't really needed here, unless you rewrite `promise.then()` into `await` call

Reply:
for some reason it prompts me to convert all the similar functions to async.
Anyways, i converted it to use async/await.